app/presenters/matrix_box_presenter.rb

```diff
@@ -51,10 +51,17 @@ def rss_log_to_presenter(rss_log)
     end
     self.time = rss_log.updated_at
 
+    # no link through if it's a carousel
+    img_link = if target.images.present? && target.images.length > 1
+                 false
+               else
+                 target.show_link_args
+               end
+
     if target&.respond_to?(:thumb_image) && target&.thumb_image
       self.image_data = {
-        image: target.thumb_image,
-        image_link: target.show_link_args,
+        images: target.images,
```

Comment:
I'm betting these have not been eager-loaded, either, since it doesn't look like you've changed anything in the controllers in this PR.

Example: for activity log, you'll need to adjust the includes at line 89 in app/controllers/rss_logs_controller.rb.  I suspect you will still need the thumbnail (you still want the user to be able to choose which is the first image of their obs users see), but you'll also need to load images:

```
   includes = {
      article: :user,
      glossary_term: :user,
      location: :user,
      name: :user,
      observation: [:location, :name, :user, **:images,**
                    @user ? { thumb_image: :image_votes } : :thumb_image],
      project: :user,
      species_list: [:location, :user]
    }
```

Reply:
Got it. Thank you. Committing some updates rn -- sorry for the premature review request.

Reply:
Quite a bit more loading. It now has to be `{ images: [:image_votes, :license, :user] }`